### PR TITLE
remove newline from printed custom type errors

### LINKED ISSUE #2827: Newlines appear in custom type errors
See `examples/failing/ProgrammableTypeErrorsTypeString.purs` for an example.

### PR DESCRIPTION
**Pull request by matthewleon**
fixes #2827

### CODE REVIEW

File: src/Language/PureScript/Errors.hs [hunk 1]
```diff
@@ -1318,7 +1319,7 @@ toTypelevelString t = (Box.text . decodeStringWithReplacement) <$> toTypelevelSt
   toTypelevelString' :: Type -> Maybe PSString
   toTypelevelString' (TypeLevelString s) = Just s
   toTypelevelString' (TypeApp (TypeConstructor f) x)
-    | f == primName "TypeString" = Just $ fromString $ prettyPrintType x
+    | f == primName "TypeString" = fromString <$> (initMay . Box.render $ typeAsBox x)
```
Comment (paf31) [inline src/Language/PureScript/Errors.hs]: Using `Box.render` might mess up indentation, and I'm not sure `initMay` is always what we want here.

Comment (matthewleon) [inline src/Language/PureScript/Errors.hs]: `prettyPrintType` was already calling `Box.render`. But yes, it's a hack. I'll think about it more.

Comment (matthewleon) [inline src/Language/PureScript/Errors.hs]: All the pretty printing operations rely on the box model. Are you suggesting coming up with an alternative pretty printer for types?

Comment (paf31) [inline src/Language/PureScript/Errors.hs]: Then that should be using `typeAsBox` actually.

Comment (matthewleon) [inline src/Language/PureScript/Errors.hs]: A bit confused by that comment. What should be using `typeAsBox`?

Comment (paf31) [inline src/Language/PureScript/Errors.hs]: Instead of `fromString . prettyPrintType` on the line above.

Comment (paf31) [inline src/Language/PureScript/Errors.hs]: In fact, the whole thing should return a `Box`.

Comment (matthewleon) [inline src/Language/PureScript/Errors.hs]: Sounds good to me. Will update it asap.

**Comment by matthewleon** [comment]
This should be much cleaner.

**Pull request merged**